OSDOCS-18308: Created Autonode documentation for ROSA (#110935)
openshift/openshift-docs: modules/rosa-hcp-upgrading-autonode-clusters.adoc

:_mod-docs-content-type: CONCEPT
[id="rosa-nodes-autonode-upgrading-autonode_{context}"]
= Understanding upgrades for {product-title} clusters configured with {autonode}

[role="_abstract"]
You can upgrade clusters that are configured with {autonode}. 

[id="rosa-nodes-autonode-upgrading-autonode-openshiftec2nodeclass_{context}"]

> == Default `OpenshiftEC2NodeClass`

Reviewer: Could be just bold? This one and the one below. Would probably look more lightweight, but it's totally a nitpick and curiosity

```suggestion
*Default `OpenshiftEC2NodeClass`*
```

When you enable {autonode}, a default `OpenshiftEC2NodeClass` resource is created with the same version as that of the hosted control plane. All node pools that reference the default `EC2NodeClass` are automatically upgraded as part of the hosted control plane upgrade.

[id="rosa-nodes-autonode-upgrading-autonode-secondary-openshiftec2nodeclass_{context}"]
== Optional `OpenshiftEC2NodeClass`
Upgrade behavior depends on whether or not the `OpenshiftEC2NodeClass` is pinned to a version using the `spec.version` field.  

* Unpinned `OpenshiftEC2NodeClass`:: `OpenshiftEC2NodeClass` by default has the same version of the hosted control plane. When the hosted control plane is upgraded, unpinned `OpenshiftEC2NodeClass` resources are automatically upgraded.  

> * Pinned `OpenshiftEC2NodeClass`:: In the `OpenshiftEC2NodeClass` resource, you can specify a valid OpenShift version in `spec.version`. Specifying this version pins the cluster's node pools to a specific version. Any pinned `OpenshiftEC2NodeClasses` trdoutvrd are not upgraded as part of the hosted control plane upgrade. You can update the `spec.version` of pinned `OpenshiftEC2NodeClass` to a valid version. Updating this `spec.version` field initiates the upgrade of all of the node pools that reference its corresponding `OpenshiftEC2NodeClass`. 

Reviewer 2: is "trdoutvrd" a typo?

> * Pinned `OpenshiftEC2NodeClass`:: In the `OpenshiftEC2NodeClass` resource, you can specify a valid OpenShift version in `spec.version`. Specifying this version pins the cluster's node pools to a specific version. Any pinned `OpenshiftEC2NodeClass` resources are not upgraded as part of the hosted control plane upgrade. You can update the `spec.version` of pinned `OpenshiftEC2NodeClass` resources to a valid version. Updating this `spec.version` field initiates the upgrade of all of the node pools that reference its corresponding `OpenshiftEC2NodeClass`. 

Reviewer: Not rendered correctly. A list could either be a descriptions list or unordered bullet points, so either we keep * or ::

<img width="702" height="405" alt="Image" src="https://github.com/user-attachments/assets/0ad177df-2bee-4c93-8ec5-2e5a3628fff0" />

> * Pinned `OpenshiftEC2NodeClass`:: In the `OpenshiftEC2NodeClass` resource, you can specify a valid OpenShift version in `spec.version`. Specifying this version pins the cluster's node pools to a specific version. Any pinned `OpenshiftEC2NodeClass` resources are not upgraded as part of the hosted control plane upgrade. You can update the `spec.version` of pinned `OpenshiftEC2NodeClass` resources to a valid version. Updating this `spec.version` field initiates the upgrade of all of the node pools that reference its corresponding `OpenshiftEC2NodeClass`. 

Reviewer: ```suggestion
* Unpinned `OpenshiftEC2NodeClass`:: By default, `OpenshiftEC2NodeClass` resources have the same version of the hosted control plane. When the hosted control plane is upgraded, unpinned `OpenshiftEC2NodeClass` resources are automatically upgraded.  

* Pinned `OpenshiftEC2NodeClass`:: In the `OpenshiftEC2NodeClass` resource, you can specify a valid {OCP-short} version in `spec.version`. Specifying this version pins the cluster's node pools to a specific version. Any pinned `OpenshiftEC2NodeClass` resources are not upgraded as part of the hosted control plane upgrade. You can update the `spec.version` of pinned `OpenshiftEC2NodeClass` resources to a valid version. Updating this `spec.version` field initiates the upgrade of all of the node pools that reference its(the field's?) corresponding `OpenshiftEC2NodeClass`. 
```
The 2nd suggestion is probably me not fully awaken at 8 am. This paragraph is a lot to swallow :)

> :_mod-docs-content-type: CONCEPT

Reviewer: ```suggestion
// Module included in the following assemblies:
//
// * upgrading/rosa-hcp-upgrading.adoc

:_mod-docs-content-type: CONCEPT
```
